src/Components/Web/src/Infrastructure/JsonOptionsServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Components.RenderTree;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Components.Web.Infrastructure;
+
+/// <summary>
+/// Extension methods for configuring web-specific JSON options for components.
+/// </summary>
+public static class JsonOptionsServiceCollectionExtensions
+{
+    /// <summary>
+    /// Configures options used for serializing JSON in web-specific components functionality.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to configure options on.</param>
+    /// <returns>The modified <see cref="IServiceCollection"/>.</returns>
+    public static IServiceCollection ConfigureComponentsWebJsonOptions(this IServiceCollection services)
+    {
+        services.ConfigureComponentsJsonOptions(static options =>
+        {
+            options.SerializerOptions.TypeInfoResolverChain.Insert(0, WebRendererJsonSerializerContext.Default);
+        });
+
+        return services;
+    }
```

Comment:
This is very confusing because its not configuring anything really (no way to pass any callback to configure anything).

Is there a reason we need some public API like this? Does this appear in user code?

Reply:
This was so that `Components.WebAssembly` and `Components.Server` could add the `WebRendererSerializerContext` to the shared `JsonOptions`. I agree that it's far from ideal, but we'll probably remove this all anyway since it seems that the feedback so far has to go with an approach similar to option 1.